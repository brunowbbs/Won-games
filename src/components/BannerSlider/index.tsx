import * as S from "./styles";

import Slider from "../Slider";
import { BannerSliderProps } from "./types";
import Banner from "../Banner";
import { Settings } from "react-slick";

const settings: Settings = {
  dots: true,
  arrows: false,
  vertical: true,
  verticalSwiping: true,
  infinite: false,
  autoplay: true,

  responsive: [
    {
      breakpoint: 1170,
      settings: {
        vertical: false,
        verticalSwiping: false,
        autoplay: true,
      },
    },
  ],
};

const BannerSlider = ({ items }: BannerSliderProps) => (
  <S.Wrapper>
    <Slider settings={settings}>
      {items.map((item) => (
        <Banner key={item.title} {...item} />
      ))}
    </Slider>
  </S.Wrapper>
);

export default BannerSlider;
