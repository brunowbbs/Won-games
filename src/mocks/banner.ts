export const bannerMock = {
  img: "https://source.unsplash.com/user/willianjusten/1042x580",
  title: "Defy death",
  subtitle: "Play the new CrashLand season",
  buttonLabel: "Buy now",
  buttonLink: "/games/defy-death",
};
