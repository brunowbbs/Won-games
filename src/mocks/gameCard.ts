export const gameCardMock = {
  title: "Population zero",
  developer: "Rockstar Games",
  img: "https://source.unsplash.com/user/willianjusten/300x144",
  price: "R$ 235,00",
  promotionalPrice: "R$ 199,00",
};
