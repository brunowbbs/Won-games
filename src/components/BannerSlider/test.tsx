import "../../../.jest/match-media-mock";

import { bannerSliderMock } from "../../mocks/bannerSlider";
import { renderWithTheme } from "../../utils/tests/helpers";

import BannerSlider from ".";
import { screen } from "@testing-library/react";

describe("<BannerSlider/>", () => {
  it("Should render vertical slider", () => {
    const { container } = renderWithTheme(
      <BannerSlider items={bannerSliderMock} />
    );

    expect(container.querySelector(".slick-vertical")).toBeInTheDocument();
  });

  it("Should render with 1 active item", () => {
    const { container } = renderWithTheme(
      <BannerSlider items={bannerSliderMock} />
    );

    expect(container.querySelectorAll(".slick-slide")).toHaveLength(3);
    expect(container.querySelectorAll("li.slick-active")).toHaveLength(1);

    expect(
      screen.getByRole("heading", { name: /defy death 1/i, hidden: false })
    ).toBeInTheDocument();

    expect(
      screen.getByRole("heading", { name: /defy death 2/i, hidden: true })
    ).toBeInTheDocument();
  });

  it("Should render with the dots", () => {
    const { container } = renderWithTheme(
      <BannerSlider items={bannerSliderMock} />
    );

    expect(container.querySelector(".slick-dots")).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
