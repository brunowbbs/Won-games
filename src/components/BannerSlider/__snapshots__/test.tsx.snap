// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerSlider/> Should render with the dots 1`] = `
<section
  class="sc-gKPRtg iAQlAu"
>
  <main
    class="sc-iBYQkv uLKON"
  >
    <div
      class="slick-slider slick-vertical slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
        style="height: 0px;"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="sc-gswNZR kkylTA"
              >
                <main
                  class="sc-bcXHqe jcpYlC"
                  color="primary"
                >
                  Bestselling
                </main>
                <div
                  aria-label="Defy death 1"
                  class="sc-dkrFOg gEZSzI"
                  role="img"
                  src="https://source.unsplash.com/user/willianjusten/1042x580"
                />
                <div
                  class="sc-hLBbgP AffZy"
                >
                  <h2
                    class="sc-eDvSVe DxYRH"
                  >
                    Defy death 1
                  </h2>
                  <h3
                    class="sc-jSUZER qrXPF"
                  >
                    Play the new crashlands season
                  </h3>
                  <a
                    class="sc-ftTHYK eEKxWj"
                    href="/games/defy-death"
                  >
                    <span>
                      Buy now
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="sc-gswNZR kkylTA"
              >
                <div
                  aria-label="Defy death 2"
                  class="sc-dkrFOg iaXmEC"
                  role="img"
                  src="https://source.unsplash.com/user/willianjusten/1042x582"
                />
                <div
                  class="sc-hLBbgP AffZy"
                >
                  <h2
                    class="sc-eDvSVe DxYRH"
                  >
                    Defy death 2
                  </h2>
                  <h3
                    class="sc-jSUZER qrXPF"
                  >
                    Play the new crashlands season
                  </h3>
                  <a
                    class="sc-ftTHYK eEKxWj"
                    href="/games/defy-death"
                  >
                    <span>
                      Buy now
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="sc-gswNZR kkylTA"
              >
                <div
                  aria-label="Defy death 3"
                  class="sc-dkrFOg PsLjt"
                  role="img"
                  src="https://source.unsplash.com/user/willianjusten/1042x581"
                />
                <div
                  class="sc-hLBbgP AffZy"
                >
                  <h2
                    class="sc-eDvSVe DxYRH"
                  >
                    Defy death 3
                  </h2>
                  <h3
                    class="sc-jSUZER qrXPF"
                  >
                    Play the new crashlands season
                  </h3>
                  <a
                    class="sc-ftTHYK eEKxWj"
                    href="/games/defy-death"
                  >
                    <span>
                      Buy now
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
      </ul>
    </div>
  </main>
</section>
`;
